Asterisk.2013/Asterisk.NET/Common.cs

```diff
@@ -26,7 +27,29 @@ public static class Common
         public static char INTERNAL_ACTION_ID_DELIMITER = '#';
 
         /// <summary> Variables delimiter </summary>
-        public static char[] VAR_DELIMITER = {'|'};
+        public static Dictionary<string, char[]> VAR_DELIMITERS = new Dictionary<string, char[]>();
+
+        /// <summary> Variables delimiter getter </summary>
+        public static char[] GET_VAR_DELIMITER(string hostname)
+        {
+            if (!VAR_DELIMITERS.ContainsKey(hostname))
+            {
+                VAR_DELIMITERS.Add(hostname, new char[] { '|' });
+            }
+
+            return VAR_DELIMITERS[hostname];
+        }
+
+        /// <summary> Variables delimiter setter </summary>
+        public static char[] SET_VAR_DELIMITER(string hostname, char[] delimiter)
+        {
+            if (!VAR_DELIMITERS.ContainsKey(hostname))

```

Comment:
You don't need this check. The dictionary indexer will do this for you, so only doing "return VAR_DELIMITERS[hostname] = delimiter;" will suffice.